src/modules/hip/kernel/color_twist.hpp

```diff
@@ -0,0 +1,339 @@
+#include <hip/hip_runtime.h>
+#include "hip/rpp_hip_common.hpp"
+#include "cpu/rpp_cpu_common.hpp"
+
+__device__ void color_twist_1RGB_hip_compute(float *pixelR, float *pixelG, float *pixelB, float4 *colorTwistParams_f4)
```

Comment:
todo:: need to do this for 4 pixels to take advantage of vector math

Reply:
@rrawther Had a discussion with Aryan on this last week - Since the algorithm takes different directions for each different RGB pixel, a vector of 8 RGBs, might take n different paths and it wouldn't make sense to vectorize the algorithm without hip versions of vector logical operations or vector comparison operations like we do for vectors on the host with _mm_cmpge_ps() (greaterthan-equal) or _mm_or_ps(). To give a resultant mask for lanes that pass or fail the condition.